src/Dialect/Krnl/KrnlHelper.cpp

```diff
@@ -519,6 +511,35 @@ void KrnlBuilder::matmul(Value A, ValueRange aStart, Value B, ValueRange bStart,
       globalUBs[1], globalUBs[2], simdize, unroll, overcompute);
 }
 
+Value KrnlBuilder::constant(
+    MemRefType type, StringRef name, DenseElementsAttr value) const {
+  static int32_t constantID = 0;
+  return b.create<KrnlGlobalOp>(loc, type, b.getI64ArrayAttr(type.getShape()),
+      b.getStringAttr(name + std::to_string(constantID++)), value,
+      /*offset=*/nullptr,
+      /*alignment=*/nullptr);
```

Comment:
@etiotto Thanks for mangling the global name! I hope we will often use this builder. It would be great if you add an optional `alignment` argument to the function too. I sometimes need to set alignment to 4K. Thanks!

Reply:
No problem @tungld. I have modified the implementation of this member function to allow optionally passing both offset and alignment.